airbyte-integrations/connectors/source-zendesk-sell/source_zendesk_sell/source.py

```diff
@@ -0,0 +1,295 @@
+from airbyte_cdk.sources.streams.http.auth import TokenAuthenticator
+from airbyte_cdk.models import SyncMode
+
+# Basic full refresh stream
+class ZendeskSellStream(HttpStream, ABC):
+    """
+    This class represents a stream output by the connector.
+    This is an abstract base class meant to contain all the common functionality at the API level e.g: the API base URL, pagination strategy,
+    parsing responses etc..
+    """
+
+    url_base = "https://api.getbase.com/v2/"
+    primary_key = None
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        try:
+            regex_page='[=?/]page[_=/-]?(\d{1,3})'
+            meta_links = response.json().get('meta', {}).get('links')
+            if 'next_page' in meta_links.keys():
+                return {'page' : int(re.findall(regex_page, meta_links['next_page'])[0])}
+        except: 
+            return None
+
+    def request_params(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        if next_page_token != None:
+            return {'page': next_page_token['page']}
+        else:
+            return {}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+
+        items = response.json()['items']
+        return [item['data'] for item in items]
+
+
+class Pipelines(ZendeskSellStream):
+    """
+    Docs: https://developer.zendesk.com/api-reference/sales-crm/resources/pipelines/
+    """
+    primary_key = "id"
+
+    def path(self, **kwargs) -> str:
+        return "pipelines"
+
+class Stages(ZendeskSellStream):
+    """
+    Docs: https://developer.zendesk.com/api-reference/sales-crm/resources/stages/
+    """
+    primary_key = "id"
+
+    def path(self, **kwargs) -> str:
+        return "stages"
+
+
+# Basic incremental stream
+class IncrementalZendeskSellStream(ZendeskSellStream, ABC):
+    state_checkpoint_interval = 100
+
+    @property
+    def cursor_field(self) -> str:
+        return "updated_at"
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Override to determine the latest state after reading the latest record. This typically compared the cursor_field from the latest record and
+        the current state and picks the 'most' recent cursor. This is how a stream's state is determined. Required for incremental.
+        """
+        if current_stream_state is not None and self.cursor_field in current_stream_state:
+            current_updated_at = current_stream_state[self.cursor_field]
+            latest_updated_at = latest_record[self.cursor_field]
+            return {self.cursor_field: max(current_updated_at, latest_updated_at)}
+        else:
+            return {self.cursor_field: '1970-01-01T00:00:00Z'}
```

Comment:
```suggestion
        if current_stream_state is not None and self.cursor_field in current_stream_state:
            current_updated_at = current_stream_state[self.cursor_field]
            latest_updated_at = latest_record[self.cursor_field]
            return {self.cursor_field: max(current_updated_at, latest_updated_at)}
        else:
            return {self.cursor_field: '1970-01-01T00:00:00Z'}
```

Can you add a unit test to validate this function.
If the `current_stream_state` is `None` this function will break because `self.cursor_field in current_stream_state (which is None)` will fail.

Also adding the `start_date` param you can change this random value of `1970-01-01...`